### PR TITLE
apis: add HostApplications for out-of-band application

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅰ. Describe what this PR does

Define APIs and configurations to identify processes for these out-of-band applications

### Ⅱ. Does this pull request fix one issue?
#1502

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 1]
```diff
@@ -317,12 +317,15 @@ type NodeSLOSpec struct {
...
 	Extensions *ExtensionsMap `json:"extensions,omitempty"`
+	// QoS management for out-of-band applications
+	HostApplications []HostApplicationSpec `json:"hostApplications,omitempty"`
 }
 
 // NodeSLOStatus defines the observed state of NodeSLO
 type NodeSLOStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+	HostApplication []HostApplicationStatus `json:"hostApplication,omitempty"`
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:328]: how about `hostApplicationStatuses`

File: apis/configuration/slo_controller_config.go [hunk 2]
```diff
@@ -88,6 +89,18 @@ type SystemCfg struct {
...
+// +k8s:deepcopy-gen=true
+type NodeHostApplicationCfg struct {
+	NodeCfgProfile `json:",inline"`
+	Applications   []slov1alpha1.HostApplicationSpec `json:"applications,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type HostApplicationCfg struct {
+	Applications   []slov1alpha1.HostApplicationSpec `json:"applications,omitempty"`
+	NodeStrategies []NodeHostApplicationCfg          `json:"nodeConfigs,omitempty"`
```
Comment (saintube) [inline apis/configuration/slo_controller_config.go:101]: perhaps
NodeStrategies `json:"nodeStrategies,omitempty"`
or
NodeConfigs `json:"nodeConfigs,omitempty"`

**Review by saintube — APPROVED** [review]
/lgtm

**Review by hormes — APPROVED** [review]

**Pull request merged**